Automattic/woocommerce-payments
# Incentives caching improvements

Fixes #6769

#### Changes proposed in this Pull Request









#### Testing instructions

1. Checkout the PR branch on your local client installation by running `git fetch && git checkout fix/6769-incentives-caching-improvements`
2. Use the WooCommerce Beta Tester plugin to set up your store for incentive eligibility
3. Make sure the incentives are cached [in the DB](http://localhost:8083/index.php?route=/table/search&db=wordpress&table=wp_options) (the `wcpay_connect_incentive` `wp_options` row) and take note of the `fetched` timestamp in the data
4. Change your store's country to a different one
5. Check that the incentives cache has been updated by looking at the `fetched` timestamp in the data
6. Refresh a WP admin page and check that the `fetched` timestamp in the data remains the same
7. Change the store's locale
8. Check that the incentives cache has been updated by looking at the `fetched` timestamp in the data. If we have translations for that locale and your store is eligible for an incentive, the incentives content should also be translated.

-------------------


**Post merge**
